builder/chroots.go

```diff
@@ -0,0 +1,637 @@
+		return fmt.Errorf("os-core bundle not found")
+	}
+
+	// Bootstrap the directories.
+	err = os.MkdirAll(filepath.Join(chrootDir, "0"), 0755)
+	if err != nil {
+		return err
+	}
+	err = os.MkdirAll(filepath.Join(outputDir, "0"), 0755)
+	if err != nil {
+		return err
+	}
+	// TODO: Move this outside this function. chroot building doesn't use LAST_VER.
+	lastVerPath := filepath.Join(chrootDir, "LAST_VER")
+	if _, err = os.Stat(lastVerPath); os.IsNotExist(err) {
+		err = ioutil.WriteFile(lastVerPath, []byte("0\n"), 0644)
+		if err != nil {
+			return err
+		}
+	}
+
+	// TODO: To not touch config code that is in flux at the moment, reparsing it here to grab
+	// information that previously Mixer didn't care about. Move that to the configuration part
+	// of Mixer.
+	cfg, err := readBuildChrootsConfig(b.Buildconf)
+	if err != nil {
+		return err
+	}
+
+	if _, ok := set.Bundles[cfg.UpdateBundle]; !ok {
+		return fmt.Errorf("couldn't find bundle %q specified in configuration as the update bundle", cfg.UpdateBundle)
+	}
+
+	// Write INI files. These are used to communicate to the next step of mixing (build update).
+	var serverINI bytes.Buffer
+	fmt.Fprintf(&serverINI, `[Server]
+emptydir=%s/empty
+imagebase=%s/image/
+outputdir=%s/www/
+`, stateDir, stateDir, stateDir)
+	if cfg.HasServerSection {
+		fmt.Fprintf(&serverINI, `
+[Debuginfo]
+banned=%s
+lib=%s
+src=%s
+`, cfg.DebugInfoBanned, cfg.DebugInfoLib, cfg.DebugInfoSrc)
+	}
+	err = ioutil.WriteFile(filepath.Join(stateDir, "server.ini"), serverINI.Bytes(), 0644)
+	if err != nil {
+		return err
+	}
+	// TODO: If we are using INI files that are case insensitive, we need to be more restrictive
+	// in bundleset to check for that. See also readGroupsINI in swupd package.
+	var groupsINI bytes.Buffer
+	for _, bundle := range set.Bundles {
+		fmt.Fprintf(&groupsINI, "[%s]\ngroup=%s\n\n", bundle.Name, bundle.Name)
+	}
+	err = ioutil.WriteFile(filepath.Join(stateDir, "groups.ini"), groupsINI.Bytes(), 0644)
+	if err != nil {
+		return err
+	}
+
+	// Mixer is used to create both Clear Linux or a mix of it.
+	var version string
+	if b.Mixver != "" {
+		fmt.Printf("Creating chroots for version %s based on Clear Linux %s\n", b.Mixver, b.Clearver)
+		version = b.Mixver
+	} else {
+		fmt.Printf("Creating chroots for version %s\n", b.Clearver)
+		version = b.Clearver
+	}
+	// TODO: This validation should happen when reading the configuration.
+	if version == "" {
+		return errors.Errorf("no Mixver or Clearver set, unable to proceed")
+	}
+
+	// TODO: Check if this is redundant with the same cleanup in builder.go!
+	chrootVersionDir := filepath.Join(chrootDir, version)
+	fmt.Printf("Removing pre-existing %s before starting\n", chrootVersionDir)
+	err = os.RemoveAll(chrootVersionDir)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Preparing new %s\n", chrootVersionDir)
+	// TODO: Fix this output when we take bundles from multiple places.
+	fmt.Printf("  based on bundles from: %s\n", b.Bundledir)
```

Comment:
This should read from .mixer/mixbundles right @kevincwells ? (once the refactor PR is in)

Reply:
Removed this print and added a print to the place we read the bundles directory. This will make the fix suggested in TODO more straightforward later.

I think the idea when Kevin changes get in is that this would not output the transient bundles directory, but some information about the original location of the bundles ("using these and these local bundles, and these and these remove bundles").